### PR TITLE
Fix #223: Update METRICS.md and README.md docs

### LINKED ISSUE #223: Update README
We haven't updated the README substantively in a few months, so it's probably a good idea to take a pass through it and make sure it accurately reflects the state of the repo now that we are close to launch.

A non-comprehensive list of changes includes:
* Instructions for enabling extraction on all sites, rather than the initial supported five:
  * For v7.0.0+, create a new string pref in `about:config` called `extensions.shopping-testpilot@mozilla.org.extractionAllowlist` and set its value to `*`.
* How to change other override prefs listed in `./src/config.js`, such as the price alert percent or absolute thresholds.
* Revise file structure section with new subfolders in `./src`.
* Consider providing static Test Pilot XPI url somewhere? https://testpilot.firefox.com/files/shopping-testpilot@mozilla.org/latest
* Add additional troubleshooting comments for building the extension from the command line re: #150 .
* Add relative URL link to `./docs/METRICS.md`
* Add instructions for using the [Fake Product Page](http://www.mkelly.me/fake-product-page/) for testing including the use of query strings to set the max price or fix the price to a particular value.
  * > Fake product page supports a static price now: http://www.mkelly.me/fake-product-page/?dollars=10000
  * > Also a configurable max value: http://www.mkelly.me/fake-product-page/?max=10000
* Link to UX mocks

### PR DESCRIPTION
**Pull request by biancadanforth**
Updating the docs a bit before launch. Took a few tries to get all the relative links working...

### CODE REVIEW

**Comment by biancadanforth** [comment]
@Osmose Ready for your review!

A couple specific questions for you:

- In `./src/utils.js`, we only use the `wait` function, not `retry`, and the `wait` function is only used in one place (`./src/background/price_updates.js`). Do we want to get rid of this `utils.js` file altogether and move `wait` into that script? I’ve left things as-is and included it in the README for now.
- We have a `"test"` command in `package.json` that doesn’t work (well, when I do `npm run test`) -- should this command be removed? We do have `pipenv run test` in the README.
- I couldn’t find a good link out for "chrome-privileged Firefox events" in the Code Organization section, so if you have a recommendation, I’m all ears!

**Comment by biancadanforth** [comment]
Also going to request an optional review from @chuckharmston .

**Comment by Osmose** [comment]
>     * In `./src/utils.js`, we only use the `wait` function, not `retry`, and the `wait` function is only used in one place (`./src/background/price_updates.js`). Do we want to get rid of this `utils.js` file altogether and move `wait` into that script? I’ve left things as-is and included it in the README for now.

Nah, we can leave it.

>     * We have a `"test"` command in `package.json` that doesn’t work (well, when I do `npm run test`) -- should this command be removed? We do have `pipenv run test` in the README.

The npm command is used by `pipenv run test`. When we actually add tests anyway that's gonna need to be refactored/removed, so just leave it be for now.

>     * I couldn’t find a good link out for "chrome-privileged Firefox events" in the Code Organization section, so if you have a recommendation, I’m all ears!

See my comment about removing that section.

**Comment by pdehaan** [comment]
FWIW, I ran a quick link checker and spell checker against the PR and nothing too alarming:

<details>
<summary><kbd>$ find . -name \*.md -not -path "./node_modules/*" -exec npx markdown-link-check {} \;</kbd></summary>

```sh
npx: installed 67 in 2.59s

FILE: ./docs/METRICS.md
[✓] ../README.md
[✓] https://github.com/erikrose/fathom
[✖] https://www.amazon.com/LEGO-Fantastic-Beasts-Grindelwald-Grindelwalds/dp/B07BKQXCZR/ref=sr_1_3_sspa?s=toys-and-games&ie=UTF8&qid=1538418041&sr=1-3-spons&keywords=legos&psc=1
[✓] https://github.com/mozilla/price-wise/issues/36#issuecomment-409641491
[✓] https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/events.html
[✓] https://bugzilla.mozilla.org/show_bug.cgi?id=1280234
[✓] https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/data/event-ping.html
[✓] https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/events.html#serialization-format
[✓] https://support.mozilla.org/en-US/kb/how-do-i-turn-do-not-track-feature
[✓] https://support.mozilla.org/en-US/kb/tracking-protection
[✓] https://support.mozilla.org/en-US/kb/disable-third-party-cookies
[✓] #common-extra-keys
[✓] https://mozilla.invisionapp.com/share/UFNSHAIMT4V#/screens/317130676_Artboard_1
[✓] https://support.mozilla.org/en-US/kb/private-browsing-use-firefox-without-history?redirectlocale=en-US&redirectslug=Private+Browsing
[✓] https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/windows/Window
[✓] https://support.mozilla.org/en-US/kb/enable-and-disable-cookies-website-preferences
[✓] https://searchfox.org/mozilla-central/source/toolkit/components/telemetry/Events.yaml
[✓] #opt-out

ERROR: dead links found!
npx: installed 67 in 2.461s

FILE: ./README.md
[✓] https://mozilla.invisionapp.com/share/UFNSHAIMT4V#/screens/317130676_Artboard_1
[✓] ./docs/METRICS.md
[✓] https://wiki.mozilla.org/Add-ons/Extension_Signing
[✓] https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/basics.html#webextensions-experiments
[✓] https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds
[✓] src/config.js
[✓] #preferences
[✓] https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/events.html
[✓] https://redux.js.org/
[✓] https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/local
[✓] https://github.com/erikras/ducks-modular-redux
[✓] https://firefox-source-docs.mozilla.org/testing/marionette/marionette/index.html
[✓] https://docs.pipenv.org/
[✓] web-ext-config.js
[✓] http://www.mkelly.me/fake-product-page/
[✓] http://www.mkelly.me/fake-product-page/?max=10000
[✓] http://www.mkelly.me/fake-product-page/?dollars=10000
[✓] https://help.github.com/articles/signing-commits/
[✖] https://circleci.com/dashboard
[✓] https://testpilot.firefox.com/files/shopping-testpilot@mozilla.org/latest
[✓] LICENSE

ERROR: dead links found!
```
</details>

I can't figure out why the amazon.com link in docs/METRICS.md is failing. It works for me if I navigate to it manually.
Also the circleci.com/dashboard page is 404ing for me, but only from a non-logged in browser. I can't imagine either of those are worth trying to fix.

---

<details>
<summary><kbd>$ npx markdown-spellcheck "README.md" "docs/*.md" -n -a -x -en-us</kbd></summary>

```sh
Spelling - README.md
See [METRICS.md](./docs/METRICS.md).
?   Add to file ignores
- A recent version of Node.js and NPM
?   Add to file ignores
g/Add-ons/Extension_Signing) [WebExtension Experiment](https://firefox-s
?   Add to file ignores
an only be run in Nightly and DevEdition with boolean `about:config` p
?   Add to file ignores
` | Compile source files with Webpack using a development configura
?   Add to file ignores
gger UI elements (such as the browserAction toolbar button) and periodica
?   Add to file ignores
_action` contains the toolbar popup for managing the list of curr
?   Add to file ignores
ins the scripts used to fetch config values specified in [config.j
?   Add to file ignores
h config values specified in [config.js](src/config.js).
?   Add to file ignores
- `src/state` contains the Redux-based code for managing global exte
?   Add to file ignores
or the add-on is managed via [Redux][]. Any time the data is chan
?   Add to file ignores
- [Pipenv][]
?   Add to file ignores
 to customize the extension's behavior for testing purposes. Some co
?   Add to file ignores
n be found in [web-ext-config.js](web-ext-config.js), and will
?   Add to file ignores
ring. List of domains (<code>{Array.string}</code> or <code>*</code>) th
?   Add to file ignores
 on all sites (default: <code>extractionAllowlist</code> value specified in [co
?   Add to file ignores
nteger. Delay before removing iframes created during price checks i
?   Add to file ignores
ag to the repo. This triggers CircleCI automation that packages, tes
?   Add to file ignores
ion is licensed under the MPL v2.0. See [`LICENSE`](LICENSE) for
?   Add to file ignores
Spelling - docs/METRICS.md
* **Product Page**: A webpage ([example](https://www.amazon
?   Add to file ignores
es**: For the initial launch (a.k.a. MVP, Minimum Viable Product)
?   Add to file ignores
uecomment-409641491): Amazon, Ebay, Walmart, Home Depot and Best
?   Add to file ignores
ent-409641491): Amazon, Ebay, Walmart, Home Depot and Best Buy.
?   Add to file ignores
* **Onboarding Popup**: The popup displayed
?   Add to file ignores
his extension, answering them thorougly requires controlled A/B testi
?   Ignore
lection/events.html) via the [WebExtensions Telemetry API](https://bugzil
?   Add to file ignores
oolbar button. One of 'add', 'price_alert', or 'none'. A value of 'unkn
?   Add to file ignores
possible if the badge text is unrecognized.
?   Add to file ignores
ends the user to a Price Wise support.mozilla.org page.
?   Add to file ignores
l, if any. One of: 'fathom', 'css_selectors', 'open_graph' or 'none'. A v
?   Add to file ignores
: 'fathom', 'css_selectors', 'open_graph' or 'none'. A value of 'none'
?   Add to file ignores
ert is dismissed (per the [UX spec](https://mozilla.invisionapp.
?   Add to file ignores
, along with all other add-on lifecycle events, is recorded by the Ad
?   Add to file ignores
le events, is recorded by the Addons Manager's event telemetry in
?   Add to file ignores
 each event is structured in [Events.yaml](https://searchfox.org/mozill
?   Add to file ignores
 page, click 'dynamic' in the dropdown menu.
?   Add to file ignores
>> 2 files are free from spelling errors
```
</details>

I think the only things I noticed were a "thorougly" typo in https://github.com/mozilla/price-wise/blob/master/docs/METRICS.md#questions-to-answer-in-future-experiments (should be "thoroughly"?)

And possibly changing "Ebay" to "eBay" in a couple places in docs/METRICS.md to match their branding/copyright:

```sh
$ git grep -n --break --heading "Ebay"

docs/METRICS.md
17:* **Supported Sites**: For the initial launch (a.k.a. MVP, Minimum Viable Product) of this extension, we are limiting the sites supported by this feature to [five websites](https://github.com/mozilla/price-wise/issues/36#issuecomment-409641491): Amazon, Ebay, Walmart, Home Depot and Best Buy.
221:  - `'ebay_link'`: Sends the user to Ebay.

src/extraction/fathom/ruleset_factory.js
160:    // Some titles (like on Ebay) are above the image, so include a top buffer
179:     * of the string (ex: 'US $5.00' on Ebay), matches any number of digits before
```

File: README.md [hunk 1]
```diff
@@ -2,6 +2,17 @@
...
 
+
+## User Experience
+
+See [UX specifications](https://mozilla.invisionapp.com/share/UFNSHAIMT4V#/screens/317130676_Artboard_1).
+
+
+## Telemetry
+
+See [METRICS.md](./docs/METRICS.md).
```
[the remark this answers is not in the document]
Comment (biancadanforth) [reply README.md]: I had to submit the PR about 3 times to get it to work, but yes!

File: docs/METRICS.md [hunk 2]
```diff
@@ -467,8 +461,9 @@ No telemetry will be sent from the extension in the following additional cases:
 - [Tracking Protection](https://support.mozilla.org/en-US/kb/tracking-protection) is enabled
   - Preference: `privacy.trackingprotection.enabled`
 - The user is in a [Private Browsing](https://support.mozilla.org/en-US/kb/private-browsing-use-firefox-without-history?redirectlocale=en-US&redirectslug=Private+Browsing)  window
-  - Preference: `browser.privatebrowsing.autostart`
   - [`windows.Window`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/windows/Window) property: `window.incognito`
+- The user has elected to [block certain kinds of cookies](https://support.mozilla.org/en-US/kb/enable-and-disable-cookies-website-preferences) and site data: all third party cookies or all cookies.
+  - Preference: `network.cookie.cookieBehavior`
```
[the remark this answers is not in the document]
Comment (biancadanforth) [reply docs/METRICS.md]: What do you mean by out of order? What kind of order are you expecting?

**Comment by biancadanforth** [comment]
@Osmose Thanks for the feedback! Ready for another look.

I wasn't sure what you meant by "out of order" in [this comment](https://github.com/mozilla/price-wise/pull/261#discussion_r232781882), so if you could clarify that, I'd be happy to make any changes there.

Thanks also @pdehaan for taking a pass! Good catch on the typo. I just removed the Amazon link to avoid any issues with that page disappearing later on from the site.

**Pull request merged**